cpp/tests/structure/induced_subgraph_test.cpp

```diff
@@ -0,0 +1,228 @@
+
+  static void SetUpTestCase() {}
+  static void TearDownTestCase() {}
+
+  virtual void SetUp() {}
+  virtual void TearDown() {}
+
+  template <typename vertex_t, typename edge_t, typename weight_t, bool store_transposed>
+  void run_current_test(
+    std::tuple<InducedSubgraph_Usecase const&, input_usecase_t const&> const& param)
+  {
+    auto [induced_subgraph_usecase, input_usecase] = param;
+
+    raft::handle_t handle{};
+    HighResClock hr_clock{};
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      hr_clock.start();
+    }
+
+    auto [graph, d_renumber_map_labels] =
+      cugraph::test::construct_graph<vertex_t, edge_t, weight_t, false, false>(
+        handle, input_usecase, induced_subgraph_usecase.test_weighted, true);
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      double elapsed_time{0.0};
+      hr_clock.stop(&elapsed_time);
+      std::cout << "construct_graph took " << elapsed_time * 1e-6 << " s.\n";
+    }
+
+    auto graph_view = graph.view();
+
+    // Construct random subgraph vertex lists
+    std::vector<size_t> h_subgraph_offsets(induced_subgraph_usecase.subgraph_sizes.size() + 1, 0);
+    std::partial_sum(induced_subgraph_usecase.subgraph_sizes.begin(),
+                     induced_subgraph_usecase.subgraph_sizes.end(),
+                     h_subgraph_offsets.begin() + 1);
+
+    rmm::device_uvector<vertex_t> all_vertices(graph_view.number_of_vertices(),
+                                               handle.get_stream());
+    cugraph::detail::sequence_fill(
+      handle.get_stream(), all_vertices.data(), all_vertices.size(), vertex_t{0});
+
+    rmm::device_uvector<vertex_t> d_subgraph_vertices(h_subgraph_offsets.back(),
+                                                      handle.get_stream());
+
+    for (size_t i = 0; i < induced_subgraph_usecase.subgraph_sizes.size(); ++i) {
+      auto start = h_subgraph_offsets[i];
+      auto last  = h_subgraph_offsets[i + 1];
+      ASSERT_TRUE(last - start <= graph_view.number_of_vertices()) << "Invalid subgraph size.";
+      // this is inefficient if last - start << graph_view.number_of_vertices() but this is for
+      // the test purpose only and the time & memory cost is only linear to
+      // graph_view.number_of_vertices(), so this may not matter.
+
+      auto vertices = cugraph::test::randomly_select(handle, all_vertices, (last - start), true);
+      raft::copy(
+        d_subgraph_vertices.data() + start, vertices.data(), vertices.size(), handle.get_stream());
+    }
+
+    auto d_subgraph_offsets = cugraph::test::to_device(handle, h_subgraph_offsets);
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      hr_clock.start();
+    }
+
+    // FIXME: turn-off do_expensive_check once verified.
```

Comment:
I guess this code is now verified, so we can turn-off expensive check and delete this FIXME.

Reply:
Turned off